File: mobile/lib/features/notification/domain/notification_repository.dart
import 'package:memory_map/features/notification/domain/notification_item.dart';

abstract interface class NotificationRepository {
  Future<List<NotificationItem>> getNotifications({int limit = 50});

  Future<int> getUnreadCount();

  Future<void> markRead(String notificationId);

  Future<void> markAllRead();
}

File: backend/src/main/java/memory_map/backend/notification/repository/NotificationRepository.java
package memory_map.backend.notification.repository;

import memory_map.backend.notification.application.NotificationReadModel;
import memory_map.backend.notification.domain.Notification;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

public interface NotificationRepository {

    void save(Notification notification);

    List<NotificationReadModel> findByRecipientUserId(
            UUID recipientUserId,
            int limit
    );

    long countUnreadByRecipientUserId(UUID recipientUserId);

    boolean markRead(UUID recipientUserId, UUID notificationId, Instant readAt);

    void markAllRead(UUID recipientUserId, Instant readAt);
}
